connection.go

```diff
@@ -250,6 +251,49 @@ func (c *Connection) shutdown() {
 	close(c.errorChan)
 }
 
+// isConnectionReset checks if an error is a connection reset error
+func (c *Connection) isConnectionReset(err error) bool {
+	errStr := err.Error()
+	return strings.Contains(errStr, "connection reset") ||
+		strings.Contains(errStr, "broken pipe")
+}
```

Comment:
We should check for error types rather than substring matches of the `Error()` output. Do we actually get any "connection closed" error that isn't `io.EOF`?